relayer/src/main.rs

```diff
@@ -94,26 +110,40 @@ async fn main() -> Result<()> {
     debug!("Established connection to Ethereum node");
 
     let config_rc2 = Arc::clone(&config);
-    let azero_connection_rc2 = Arc::clone(&azero_connection);
-    let eth_connection_rc2 = Arc::clone(&eth_connection);
-    let redis_connection_rc2 = Arc::clone(&redis_connection);
+    let azero_signed_connection_rc1 = Arc::clone(&azero_signed_connection);
+    let eth_connection_rc1 = Arc::clone(&eth_connection);
+    let redis_connection_rc1 = Arc::clone(&redis_connection);
+    let emergency_rc2 = Arc::clone(&emergency);
 
     info!("Starting Ethereum listener");
 
     tasks.spawn(async move {
-        EthListener::run(config, azero_connection, eth_connection, redis_connection)
-            .await
-            .map_err(ListenerError::from)
+        EthListener::run(
+            config_rc2,
+            azero_signed_connection_rc1,
+            eth_connection_rc1,
+            redis_connection_rc1,
+            emergency_rc2,
```

Comment:
Using rc numbers starts to become confusing when we pass arcs with different counters to the same task.
Maybe we should suffix with the name of corresponding task and not just a number (or somehow make it uniform within one spawn, number in the opposite direction for example)

Reply:
done